src/overlays/actors/ovl_Obj_Purify/z_obj_purify.c

```diff
@@ -27,42 +43,252 @@ const ActorInit Obj_Purify_InitVars = {
+        Actor_SetObjectDependency(play, &this->dyna.actor);
+        if (sp28 == 1) {
+            DynaPolyActor_LoadMesh(play, &this->dyna, ObjPurifyInfoList[sp24].colHeader);
+            func_80A84FA0(this);
+        } else if (ObjPurify_IsPurified(this)) {
+            func_80A8515C(this);
+        } else {
+            func_80A84FEC(this);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Purify/func_80A84CD0.s")
+void func_80A84FA0(ObjPurify* this) {
+    this->actionFunc = func_80A84FB4;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Purify/func_80A84CF8.s")
+void func_80A84FB4(ObjPurify* this, PlayState* play) {
+    if (ObjPurify_IsPurified(this)) {
+        Actor_MarkForDeath(&this->dyna.actor);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Purify/ObjPurify_Init.s")
+void func_80A84FEC(ObjPurify* this) {
+    this->dyna.actor.draw = func_80A851C8;
+    this->gfxIndex = 0;
+    this->actionFunc = func_80A85010;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Purify/ObjPurify_Destroy.s")
+void func_80A85010(ObjPurify* this, PlayState* play) {
+    if (ObjPurify_IsPurified(this)) {
+        func_80A85048(this);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Purify/func_80A84EAC.s")
+void func_80A85048(ObjPurify* this) {
+    this->dyna.actor.draw = func_80A851C8;
+    this->gfxIndex = 0;
+    this->actionFunc = func_80A85074;
+    this->timer = 30;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Purify/func_80A84EC0.s")
+void func_80A85074(ObjPurify* this, PlayState* play) {
+    this->timer--;
+    if (this->timer <= 0) {
+        func_80A850B0(this);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Purify/func_80A84FA0.s")
+void func_80A850B0(ObjPurify* this) {
+    f32 ypos = this->dyna.actor.home.pos.y;
+    this->dyna.actor.draw = func_80A85304;
+    this->unk168[1] = ypos - 90.0f;
+    this->actionFunc = func_80A850E8;
+    this->unk168[0] = ypos;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Purify/func_80A84FB4.s")
+void func_80A850E8(ObjPurify* this, PlayState* play) {
+    Math_StepToF(&this->unk168[0], this->dyna.actor.home.pos.y - 90.0f, 0.9f);
+    if (Math_StepToF(&this->unk168[1], this->dyna.actor.home.pos.y, 0.9f)) {
+        func_80A8515C(this);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Purify/func_80A84FEC.s")
+void func_80A8515C(ObjPurify* this) {
+    this->dyna.actor.flags &= ~0x10;
+    this->dyna.actor.draw = func_80A851C8;
+    this->gfxIndex = 1;
+    this->actionFunc = ObjPurify_DoNothing;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Purify/func_80A85010.s")
+void ObjPurify_DoNothing(ObjPurify* this, PlayState* play) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Purify/func_80A85048.s")
+void ObjPurify_Update(Actor* thisx, PlayState* play) {
+    ObjPurify* this = THIS;
```

Comment:
```suggestion
    ObjPurify* this = THIS;

```
blank line after declarations.